lib/backend/block_affinity.go

```diff
@@ -44,7 +44,19 @@ func (key BlockAffinityKey) asEtcdKey() (string, error) {
 }
 
 func (key BlockAffinityKey) asEtcdDeleteKey() (string, error) {
-	return key.asEtcdKey()
+	// Must specify at least a host to remove.
+	if key.Host == "" {
+		return "", common.ErrorInsufficientIdentifiers{}
+	}
+	k := "/calico/ipam/v2/host/" + key.Host
```

Comment:
One other thing - provided it's been merged, the insufficient indices error now accepts the field name that is missing.